Ambient occlusion

Nika-Keks opened the pull request.

Nika-Keks committed:
added ambient occlusion render

Nika-Keks committed:
added ambient occlusion test

DariaWelt commented on pyrt/renderer/__init__.py:
```diff
@@ -8,3 +8,4 @@
 from .simplert import SimpleRT
 from .mprt import MultiProcessRT
 from .rgbimage import RGBImage, loadimage
+from .aosimplert import AOSimpleRT
```
No newline at end of file

DariaWelt commented on test-features.ipynb:
```diff
@@ -503,6 +503,159 @@
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
```
I think we should remove the output in this file to be consistent with the source

Nika-Keks replied:
I agree, I'll fix it later

DariaWelt reviewed (commented):
Good solution.

Commit:
Added example of AORenderer

derkanw commented on examples/20_ambient_occlusion.py:
```diff
@@ -0,0 +1,56 @@
+scene.add(Sphere(center=Vec3(-2, 1, 1), radius=1, material=sphere_material))
+
+scene.add(Sphere(center=Vec3(-1, -1, 1), radius=1, material=sphere_material))
+scene.add(Sphere(center=Vec3(1, -1, 1), radius=1, material=sphere_material))
```
Очень хочется это сделать циклом

derkanw commented on pyrt/renderer/aosimplert.py:
```diff
@@ -0,0 +1,45 @@
+        super().__init__(shadow, iterations)
+        
+        self.aorays_len = aoray_len
+        self.random_directions = [Vec3(*((random.random()) * 2 - 1 for _ in range(3))) for _ in range(n_aorays)]
```
Не очень поняла, почему координаты вектора подсчитываются именно так: (random.random()) * 2 - 1

Nika-Keks committed:
fixed bugs
 - added endl in render/__init__
 - cleaned off output in test-feature.ipynd
 - rewrote random generation

Nika-Keks committed:
cleared test-features.ipynd
